src/fswAlgorithms/attGuidance/oneAxisSolarArrayPoint/oneAxisSolarArrayPoint.c

```diff
@@ -48,7 +48,52 @@ void SelfInit_oneAxisSolarArrayPoint(oneAxisSolarArrayPointConfig *configData, i
 */
 void Reset_oneAxisSolarArrayPoint(oneAxisSolarArrayPointConfig *configData, uint64_t callTime, int64_t moduleID)
 {
+    if (!NavAttMsg_C_isLinked(&configData->attNavInMsg)) {
+        _bskLog(configData->bskLogger, BSK_ERROR, " oneAxisSolarArrayPoint.attNavInMsg wasn't connected.");
+    }
 
+    // check how the input body heading is provided
+    if (BodyHeadingMsg_C_isLinked(&configData->bodyHeadingInMsg)) {
+        configData->bodyAxisInput = inputBodyHeadingMsg;
+    }
+    else {
+        if (v3Norm(configData->h1Hat_B) > EPS) {
+            configData->bodyAxisInput = inputBodyHeadingParameter;
+        }
+        else {
+            _bskLog(configData->bskLogger, BSK_ERROR, " oneAxisSolarArrayPoint.bodyHeadingInMsg wasn't connected and no body heading h1Hat_B was specified.");
+        }
+    }
+
+    // check how the input inertial heading is provided
+    if (InertialHeadingMsg_C_isLinked(&configData->inertialHeadingInMsg) || EphemerisMsg_C_isLinked(&configData->ephemerisInMsg)) {
```

Comment:
:small_red_triangle_down: This logic can be simplified. By my read the first conditional is redundant. It's implicitly checked by the subsequent two conditionals. Do you agree?

Reply:
I've removed one nested if statement, while checking all the different cases. Is that what you wanted?